src/kernel/panic.zig

```diff
@@ -1,13 +1,382 @@
+        var it = std.debug.StackIterator.init(first_ret_addr);
+        while (it.next()) |ret_addr| {
+            if (ret_addr != last_addr) logTraceAddress(ret_addr);
+            last_addr = ret_addr;
+        }
+    }
     arch.haltNoInterrupts();
 }
+
+///
+/// Parse a hexadecimal address from the pointer up until the end pointer. Must be terminated by a whitespace character.
+///
+/// Arguments:
+///     INOUT ptr: *[*]const u8 - The address at which to start looking, updated after all characters have been consumed.
+///     IN: end: *const u8 - The end address at which to start looking. A whitespace character must be found before this.
+///
+/// Return: The address parsed.
+///
+/// Error: PanicError || std.fmt.ParseUnsignedError
+///     PanicError.InvalidSymbolFile: A terminating whitespace wasn't found before the end address.
+///     std.fmt.ParseUnsignedError: See std.fmt.parseInt
+///
+fn parseAddr(ptr: *[*]const u8, end: *const u8) !usize {
+    const addr_start = ptr.*;
+    ptr.* = try parseNonWhitespace(ptr.*, end);
+    const len = @ptrToInt(ptr.*) - @ptrToInt(addr_start);
+    const addr_str = addr_start[0..len];
+    return try std.fmt.parseInt(usize, addr_str, 16);
+}
+
+///
+/// Parse a single character. The address given cannot be greater than or equal to the end address given.
+///
+/// Arguments:
+///     IN ptr: [*]const u8 - The address at which to get the character from.
+///     IN: end: *const u8 - The end address at which to start looking. ptr cannot be greater than or equal to this.
+///
+/// Return: The character parsed.
+///
+/// Error: PanicError
+///     PanicError.InvalidSymbolFile: The address given is greater than or equal to the end address.
+///
+fn parseChar(ptr: [*]const u8, end: *const u8) PanicError!u8 {
+    if (@ptrToInt(ptr) >= @ptrToInt(end)) return PanicError.InvalidSymbolFile;
+    return ptr[0];
+}
+
+///
+/// Parse until a non-whitespace character. Must be terminated by a non-whitespace character before the end address.
+///
+/// Arguments:
+///     IN ptr: [*]const u8 - The address at which to start looking.
+///     IN: end: *const u8 - The end address at which to start looking. A non-whitespace character must be found before this.
+///
+/// Return: The number of whitespace characters consumed.
+///
+/// Error: PanicError
+///     PanicError.InvalidSymbolFile: A terminating non-whitespace character wasn't found before the end address.
+///
+fn parseWhitespace(ptr: [*]const u8, end: *const u8) PanicError![*]const u8 {
+    var i: u32 = 0;
+    while (std.fmt.isWhiteSpace(try parseChar(ptr + i, end))) : (i += 1) {}
+    return ptr + i;
+}
+
+///
+/// Parse until a whitespace character. Must be terminated by a whitespace character before the end address.
+///
+/// Arguments:
+///     IN ptr: [*]const u8 - The address at which to start looking.
+///     IN: end: *const u8 - The end address at which to start looking. A whitespace character must be found before this.
```

Comment:
You have colon that you don't need (after the IN). Also found in a couple of places